src-tauri/src/terminal.rs

```diff
@@ -0,0 +1,414 @@
+                let mut cmd = Command::new("lxterminal");
+                cmd.arg("--version").output().await.is_ok()
+            }
+            Terminal::Tilix => {
+                let mut cmd = Command::new("tilix");
+                cmd.arg("--version").output().await.is_ok()
+            }
+            Terminal::DeepinTerminal => {
+                let mut cmd = Command::new("deepin-terminal");
+                cmd.arg("--version").output().await.is_ok()
+            }
+            Terminal::CoolRetroTerm => {
+                let mut cmd = Command::new("cool-retro-term");
+                cmd.arg("--version").output().await.is_ok()
+            }
+            Terminal::Alacritty => {
+                let mut cmd = Command::new("alacritty");
+                cmd.arg("--version").output().await.is_ok()
+            }
+            Terminal::Terminal => { get_os() == "macos" }
+        }
+    }
+    //endregion
+
+
+    //region Running a command in the terminal
+    /**
+    Returns a [`Command`] that, when executed should open the terminal and run the command.
+
+
+    ## Commands
+    `{command}` = `{command};echo Command finished with code $?;sleep infinity`
+
+    | Terminal         | Command to open terminal                                                 |
+    |------------------|--------------------------------------------------------------------------|
+    | cmd              | `start cmd.exe /k  {command}`                                            |
+    | GNOMETerminal    | `gnome-terminal -- /usr/bin/env sh -c  {command}`                        |
+    | GNOMEConsole     | `kgx -e /usr/bin/env sh -c  {command}`                                   |
+    | Konsole          | `konsole -e /usr/bin/env sh -c  {command}`                               |
+    | Xfce4Terminal    | `xfce4-terminal -x /usr/bin/env sh -c  {command}`                        |
+    | Terminator       | `terminator -T "Downloading depot..." -e  {command}`                     |
+    | Terminology      | `terminology -e /usr/bin/env sh -c  {command}`                           |
+    | XTerm            | `xterm -hold -T "Downloading depot..." -e /usr/bin/env sh -c  {command}` |
+    | Kitty            | `kitty /usr/bin/env sh -c  {command}`                                    |
+    | LXTerminal       | `lxterminal -e /usr/bin/env sh -c  {command}`                            |
+    | Tilix            | `tilix -e /usr/bin/env sh -c  {command}`                                 |
+    | DeepinTerminal   | `deepin-terminal -e /usr/bin/env sh -c  {command}`                       |
+    | CoolRetroTerm    | `cool-retro-term -e /usr/bin/env sh -c  {command}`                       |
+    | Alacritty        | `alacritty -e /usr/bin/env sh -c  {command}`                             |
+    | Terminal (macOS) | We create a bash script and run that using `open`.                       |
+
+     */
+    pub fn create_command(&self, steam_download: &SteamDownload) -> Command {
```

Comment:
same story here. 
maybe you could create a config or terminals file that has a array of terminal dicts/objects with all the info needed for that terminal.

should keep everything in sync as it is easy to forget to update a terminal somewhere with this many places. 
the gui could also get it's options from there.
Would make the code a whole lot simpeler and terminals could easily be updated/added/removed.